gammapy/catalog/hawc.py

```diff
@@ -196,7 +169,114 @@ class SourceCatalog2HWC(SourceCatalog):
     description = "2HWC catalog from the HAWC observatory"
     """Catalog description"""
 
-    source_object_class = SourceCatalogObject2HWC
+    source_object_class = SourceCatalogObjectHWC
+
+    def __init__(self, filename="$GAMMAPY_DATA/catalogs/2HWC.ecsv"):
+        table = Table.read(make_path(filename), format="ascii.ecsv")
+
+        source_name_key = "source_name"
+
+        super().__init__(table=table, source_name_key=source_name_key)
+
+
+
+
+class SourceCatalogObject2HWC(SourceCatalogObjectHWC):
+
+
+    def spectral_model(self, which="point"):
+        """Spectral model (`~gammapy.modeling.models.PowerLawSpectralModel`).
+
+        * ``which="point"`` -- Spectral model under the point source assumption.
+        * ``which="extended"`` -- Spectral model under the extended source assumption.
+          Only available for some sources. Raise ValueError if not available.
+        """
+        idx = self._get_idx(which)
+
+        pars = {
+            "reference": "7 TeV",
+            "amplitude": self.data[f"spec{idx}_dnde"],
+            "index": -self.data[f"spec{idx}_index"],
+        }
+
+        errs = {
+            "amplitude": self.data[f"spec{idx}_dnde_err"],
+            "index": self.data[f"spec{idx}_index_err"],
+        }
+
+        model = Model.create("PowerLawSpectralModel", "spectral", **pars)
+
+        for name, value in errs.items():
+            model.parameters[name].error = value
+
+        return model
+
+
+class SourceCatalogObject3HWC(SourceCatalogObjectHWC):
+
+
+    def spectral_model(self, which="point"):
+        """Spectral model (`~gammapy.modeling.models.PowerLawSpectralModel`).
+
+        * ``which="point"`` -- Spectral model under the point source assumption.
+        * ``which="extended"`` -- Spectral model under the extended source assumption.
+          Only available for some sources. Raise ValueError if not available.
+        """
+        # index_errp = self.data["spec_{idx}_index_errp"]
+        # index_errn = self.data["spec_{idx}_index_errn"]
+        # index_err = 0.5 * (index_errn + index_errp)
+        #
+        #
+        #
+        # amplitude_errp = self.data["spec_{idx}_dnde_errp"]
+        # amplitude_errn = self.data["spec_{idx}_dnde_errp"]
+        # amplitude_err = 0.5 * (index_errn + index_errp)
+
+        idx = self._get_idx(which)
+
+        pars = {
+            "reference": "7 TeV",
+            "amplitude": self.data[f"spec{idx}_dnde"],
+            "index": -self.data[f"spec{idx}_index"],
+        }
+
+        errs = {
+             "index_err" : 0.5 * ( self.data[f"spec_{idx}_index_errp"] + self.data[f"spec_{idx}_index_errn"]),
+             "amplitude_err" : 0.5 * (self.data[f"spec_{idx}_dnde_errp"] + self.data[f"spec_{idx}_dnde_errp"])
+        }
+
+        model = Model.create("PowerLawSpectralModel", "spectral", **pars)
+
+        for name, value in errs.items():
+            model.parameters[name].error = value
+
+        return model
+
+class SourceCatalog3HWC():
```

Comment:
HEre you have to add the inheritance from `SourceCatalog`

Reply:
done